src/lib/lib.gni

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+declare_args() {
+  # Enable libshell support.
+  chip_build_libshell = false
```

Comment:
This seems fine, but should this default to `true` on Linux to preserve current behavior?

Reply:
I'm fine to enable it back into `examples/platform/linux/BUILD.gn` for `Linux` platform if required. I have disabled it mostly because `chip_build_libshell` was `false` by default in `src/lib/BUILD.gn` and other platforms enable it conditionally.
Furthermore, I've tried to see what it does by typing commands in and there is not much commands at the moment, so I'm not sure how much it is used.  Again if someone insist, I will add it back.


For references the commands are: 
```
help
help
  base64          Base64 encode / decode utilities
  exit            Exit the shell application
  help            List out all top level commands
  version         Output the software version
  config          Dump device configuration. Usage: config [param_name]
Done
```